test/e2e/network/networking_perf.go

```diff
@@ -160,3 +172,275 @@ var _ = SIGDescribe("Networking IPerf [Experimental] [Slow] [Feature:Networking-
+//   data to the server and back for ten seconds, after which the results are collected and parsed.
+//   Thus, if your cluster has 10 nodes, then 10 test runs are performed.
+//     Note: a more complete test could run this scenario with a daemonset of servers as well; however, this
+//     would require n^2 tests, n^2 time, and n^2 network resources which quickly become prohibitively large
+//     as the cluster size increases.
+//   Finally, after collecting all data, the results are analyzed and tabulated.
+/*
+iperf2 data looks like:
+
+------------------------------------------------------------
+Client connecting to iperf2-server, TCP port 6789 with pid 15
+TCP window size: 85.0 KByte (default)
+------------------------------------------------------------
+[  3] local 10.244.1.45 port 57172 connected with 10.96.114.79 port 6789
+[ ID] Interval        Transfer    Bandwidth       Write/Err
+[  3] 0.00-1.00 sec  2.20 GBytes  18.9 Gbits/sec  18044/0
+[  3] 1.00-2.00 sec  2.38 GBytes  20.5 Gbits/sec  19519/0
+[  3] 2.00-3.00 sec  2.59 GBytes  22.2 Gbits/sec  21188/0
+[  3] 3.00-4.00 sec  2.32 GBytes  19.9 Gbits/sec  18995/0
+[  3] 4.00-5.00 sec  2.42 GBytes  20.8 Gbits/sec  19822/0
+[  3] 5.00-6.00 sec  2.35 GBytes  20.2 Gbits/sec  19247/0
+[  3] 6.00-7.00 sec  2.43 GBytes  20.8 Gbits/sec  19867/0
+[  3] 7.00-8.00 sec  2.25 GBytes  19.4 Gbits/sec  18470/0
+[  3] 8.00-9.00 sec  2.52 GBytes  21.7 Gbits/sec  20683/0
+[  3] 0.00-9.97 sec  23.5 GBytes  20.2 Gbits/sec  192361/0
+
+Or in CSV format:
+
+20201210141800.884,10.244.2.24,47880,10.96.114.79,6789,3,0.0-1.0,1677852672,13422821376
+20201210141801.881,10.244.2.24,47880,10.96.114.79,6789,3,1.0-2.0,1980760064,15846080512
+20201210141802.883,10.244.2.24,47880,10.96.114.79,6789,3,2.0-3.0,1886650368,15093202944
+20201210141803.882,10.244.2.24,47880,10.96.114.79,6789,3,3.0-4.0,2035417088,16283336704
+20201210141804.879,10.244.2.24,47880,10.96.114.79,6789,3,4.0-5.0,1922957312,15383658496
+20201210141805.881,10.244.2.24,47880,10.96.114.79,6789,3,5.0-6.0,2095316992,16762535936
+20201210141806.882,10.244.2.24,47880,10.96.114.79,6789,3,6.0-7.0,1741291520,13930332160
+20201210141807.879,10.244.2.24,47880,10.96.114.79,6789,3,7.0-8.0,1862926336,14903410688
+20201210141808.878,10.244.2.24,47880,10.96.114.79,6789,3,8.0-9.0,1821245440,14569963520
+20201210141809.849,10.244.2.24,47880,10.96.114.79,6789,3,0.0-10.0,18752208896,15052492511
+*/
+// TODO remove other tests in this file if this test proves to be more stable, informative, and useful
+var _ = SIGDescribe("Networking IPerf2", func() {
+	// this test runs iperf2: one pod as a server, and a daemonset of clients
+	f := framework.NewDefaultFramework("network-perf")
+
+	ginkgo.It(fmt.Sprintf("should run iperf2"), func() {
+		nodeList, err := e2enode.GetReadySchedulableNodes(f.ClientSet)
+		framework.ExpectNoError(err)
+
+		serverPodsListOptions := metav1.ListOptions{
+			LabelSelector: "app=iperf2-server",
```

Comment:
you should pass the labels as a parameter to the functions that creates the deployments and daemonsets, ...., otherwise you are defining them in 2 places and it can be easy to go out of sync

Reply:
👍